src/site/content/en/blog/designcember-calculator/index.md

```diff
@@ -0,0 +1,300 @@
+          return response;
+        }
+        return fetch(event.request);
+      } catch {
+        return new Response('Offline');
+      }
+    })(),
+  );
+});
+```
+
+## Blending in on mobile
+
+Now that the app is installable, the next step is to make it blend in as much as possible. On
+mobile, this can be done by setting the display mode to `fullscreen` in the Web App Manifest.
+
+```json
+{
+  "display": "fullscreen"
+}
+```
+
+On devices with a camera hole or notch,
+[tweaking the viewport](https://webkit.org/blog/7929/designing-websites-for-iphone-x/) so that the
+content covers the whole screen makes the app look gorgeous.
+
+```html
+<meta name="viewport" content="initial-scale=1, viewport-fit=cover" />
+```
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/kEOrYPiDbg6DSTGHeB0C.jpg", alt="Designcember Calculator running fullscreen on a Pixel 6 Pro phone.", width="800", height="773" %}
+
+## Blending in on desktop
+
+On desktop, there is a cool feature I can make use of,
+[Window Controls Overlay](/window-controls-overlay/), which allows me to put content in the title
+bar of the app window. The first step is to override the display mode fallback sequence so it tries
+to use `window-controls-overlay` first when it is available.
+
+```json
+{
+  "display_override": ["window-controls-overlay"]
+}
+```
+
+This makes the title bar effectively go away and the content moves up into the title bar area as if
+the title bar were not there. My idea is to move the skeuomorphic solar cell up into the title bar
+and the rest of the calculator UI down accordingly, which I can do with some CSS that makes use of
+the `titlebar-area-*` environment variables. You will notice that all the selectors carry a `wco`
+class, which will be relevant a couple of paragraphs down.
+
+```css
+#calc_solar_cell.wco {
+  position: fixed;
+  left: calc(0.25rem + env(titlebar-area-x, 0));
+  top: calc(0.75rem + env(titlebar-area-y, 0));
+  width: calc(env(titlebar-area-width, 100%) - 0.5rem);
+  height: calc(env(titlebar-area-height, 33px) - 0.5rem);
+}
+
+#calc_display_surface.wco {
+  margin-top: calc(env(titlebar-area-height, 33px) - 0.5rem);
+}
+```
+
+Up next, I need to decide which elements to make draggable, since the title bar that I would usually
+use for dragging is not available. In the style of a classic widget, I can even decide to make the
+entire calculator draggable by applying `(-webkit-)app-region: drag`, apart from the buttons, which
+get `(-webkit-)app-region: no-drag` so they cannot be used to drag.
+
+```css
+#calc_inside.wco,
+#calc_solar_cell.wco {
+  -webkit-app-region: drag;
+  app-region: drag;
+}
+
+button {
+  -webkit-app-region: no-drag;
+  app-region: no-drag;
+}
+```
+
+The final step is to make the app reactive to window controls overlay changes. In a true progressive
+enhancement approach, I only load the code for this feature when the browser supports it.
+
+```js
+if ('windowControlsOverlay' in navigator) {
+  import('/wco.js');
```

Comment:
Have we ever written an article about this?

Reply:
Yes, it's linked on line 104: https://web.dev/window-controls-overlay/.